src/EditorFeatures/CSharp/Wrapping/SeparatedSyntaxList/CSharpArgumentWrapper.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Editor.Wrapping.SeparatedSyntaxList
+{
+    internal partial class CSharpArgumentWrapper
+        : AbstractCSharpSeparatedSyntaxListWrapper<BaseArgumentListSyntax, ArgumentSyntax>
+    {
+        protected override string ListName => FeaturesResources.argument_list;
+        protected override string ItemNamePlural => FeaturesResources.arguments;
+        protected override string ItemNameSingular => FeaturesResources.argument;
+
+        protected override SeparatedSyntaxList<ArgumentSyntax> GetListItems(BaseArgumentListSyntax listSyntax)
+            => listSyntax.Arguments;
+
+        protected override BaseArgumentListSyntax GetApplicableList(SyntaxNode node)
+        {
+            switch (node)
+            {
+                case InvocationExpressionSyntax invocationExpression: return invocationExpression.ArgumentList;
+                case ElementAccessExpressionSyntax elementAccessExpression: return elementAccessExpression.ArgumentList;
+                case ObjectCreationExpressionSyntax objectCreationExpression: return objectCreationExpression.ArgumentList;
+                case ConstructorInitializerSyntax constructorInitializer: return constructorInitializer.ArgumentList;
+            }
+
+            return null;
+        }
+
+        protected override bool PositionIsApplicable(
+            SyntaxNode root, int position,
+            SyntaxNode declaration, BaseArgumentListSyntax listSyntax)
+        {
+            var startToken = listSyntax.GetFirstToken();
+
+            if (declaration is InvocationExpressionSyntax ||
```

Comment:
so, it handles comments, or preprocessor in between arguments/binary operator/parameters? and also, can handle argument being multiline expression including lambda and etc?

Reply:
No.  As mentioned in the PR and in the code:

> We don't offer this if any items in the list have non-whitespace leading/trailing trivia. It is fairly unclear what should happen with those items, so for now we just do not offer. If this is important, we can come up with a strategy on how to handle these.

> We don't offer this if any items in the list are multiline. With multiline, if you move the items around, you can really mess up formatting of the information inside. If this is important, we can come up with a strategy on how to handle these.